{
    "version": "1.0.0",
    "title": "System is resilient to provider's failures",
    "description": "Can our consumer survive gracefully a provider's failure?",
    "tags": [
        "service",
        "kubernetes",
        "quarkus"
    ],
    "steady-state-hypothesis": {
        "title": "Services are all available and healthy",
        "probes": [
            {
                "type": "probe",
                "name": "consumer-service-must-still-respond",
                "tolerance": 200,
                "provider": {
                    "type": "http",
                    "url": "http://192.168.64.39:32568/hello"
                }
            }
        ]
    },
    "method": [
        {
            "type": "action",
            "name": "terminate-pods",
            "provider": {
                "type": "python",
                "module": "chaosk8s.pod.actions",
                "func": "terminate_pods",
                "arguments": {
                    "label_selector": "name=my-provider-app",
                    "rand": true,
                    "ns": "default"
                }
            }
        },
        {
            "type": "probe",
            "name": "consumer-service-must-still-respond",
            "tolerance": 200,
            "provider": {
                "type": "http",
                "url": "http://192.168.64.39:32568/hello"
            }
        }
    ],
    "rollbacks": []
}